checkov/common/graph/graph_builder/local_graph.py

```diff
@@ -20,3 +19,63 @@ def __init__(self) -> None:
     @abstractmethod
     def build_graph(self, render_variables: bool) -> None:
         pass
+
+    def get_vertices_with_degrees_conditions(
+        self, out_degree_cond: Callable[[int], bool], in_degree_cond: Callable[[int], bool]
+    ) -> List[int]:
+        vertices_with_out_degree = {
+            vertex_index for vertex_index in self.out_edges.keys() if out_degree_cond(len(self.out_edges[vertex_index]))
+        }
+        vertices_with_in_degree = {
+            vertex_index for vertex_index in self.in_edges.keys() if in_degree_cond(len(self.in_edges[vertex_index]))
+        }
+
+        return list(vertices_with_in_degree.intersection(vertices_with_out_degree))
+
+    def get_in_edges(self, end_vertices: List[int]) -> List[Edge]:
+        res = []
+        for vertex in end_vertices:
+            res.extend(self.in_edges.get(vertex, []))
+        return self.sort_edged_by_dest_out_degree(res)
+
+    def sort_edged_by_dest_out_degree(self, edges: List[Edge]) -> List[Edge]:
+        edged_by_out_degree: Dict[int, List[Edge]] = {}
+        for edge in edges:
+            dest_out_degree = len(self.out_edges[edge.dest])
+            edged_by_out_degree.setdefault(dest_out_degree, []).append(edge)
+        sorted_edges = []
+        for degree in sorted(edged_by_out_degree.keys()):
+            sorted_edges.extend(edged_by_out_degree[degree])
+        return sorted_edges
```

Comment:
when I'm not mistaken, then you can simplify by using `sorted(edges, key=...)`. Probably worth a try 😉